jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/skaffold/FilesMojoV2.java

```diff
@@ -0,0 +1,207 @@
+          .map(Paths::get)
+          .forEach(skaffoldFilesOutput::addInput);
+
+      // This seems weird, but we will only print out the jib "extraFiles" directory on projects
+      // where the plugin is explicitly configured (even though _skaffold-files-v2 is a
+      // jib-maven-plugin goal and is expected to run on all projects irrespective of their
+      // configuring of the jib plugin).
+      if (project.getPlugin(MavenProjectProperties.PLUGIN_KEY) != null) {
+        // Add extra directory
+        skaffoldFilesOutput.addInput(resolveExtraDirectory(project));
+      }
+
+      // Grab non-project SNAPSHOT dependencies for this project
+      // TODO: this whole sections relies on internal maven API, it could break. We need to explore
+      // TODO: better ways to resolve dependencies using the public maven API.
+      Set<String> projectArtifacts =
+          projects
+              .stream()
+              .map(MavenProject::getArtifact)
+              .map(Artifact::toString)
+              .collect(Collectors.toSet());
+
+      DependencyFilter ignoreProjectDependenciesFilter =
+          (node, parents) -> {
+            if (node == null || node.getDependency() == null) {
+              // if nothing, then ignore
+              return false;
+            }
+            if (projectArtifacts.contains(node.getArtifact().toString())) {
+              // ignore project dependency artifacts
+              return false;
+            }
+            // we only want compile/runtime deps
+            return Artifact.SCOPE_COMPILE_PLUS_RUNTIME.contains(node.getDependency().getScope());
+          };
+
+      try {
+        DependencyResolutionResult resolutionResult =
+            projectDependenciesResolver.resolve(
+                new DefaultDependencyResolutionRequest(project, session.getRepositorySession())
+                    .setResolutionFilter(ignoreProjectDependenciesFilter));
+        resolutionResult
+            .getDependencies()
+            .stream()
+            .map(org.eclipse.aether.graph.Dependency::getArtifact)
+            .filter(org.eclipse.aether.artifact.Artifact::isSnapshot)
+            .map(org.eclipse.aether.artifact.Artifact::getFile)
+            .map(File::toPath)
+            .forEach(skaffoldFilesOutput::addInput);
+
+      } catch (DependencyResolutionException ex) {
+        throw new MojoExecutionException("Failed to resolve dependencies", ex);
+      }
+    }
+
+    try {
+      // Print JSON string
+      System.out.println("BEGIN JIB JSON");
+      System.out.println(skaffoldFilesOutput.getJsonString());
+      System.out.println("END JIB JSON");
+    } catch (IOException ex) {
+      throw new MojoExecutionException(ex.getMessage(), ex);
+    }
+  }
+
+  private Path resolveExtraDirectory(MavenProject project) {
+    // Try getting extra directory from project/session properties
+    if (project.getProperties().containsKey(PropertyNames.EXTRA_DIRECTORY_PATH)) {
```

Comment:
I wonder if we can make this a static method in the project properties file (it's the same code?)